crates/jit/src/lib.rs

```diff
@@ -22,16 +22,23 @@
 
 #[cfg(feature = "parallel-compilation")]
 macro_rules! maybe_parallel {
-    ($e:ident.($serial:ident | $parallel:ident)) => {
-        $e.$parallel()
-    };
+   ($condition:ident, $e:ident.($serial:ident | $parallel:ident), $iter_name:ident => { $body:expr }) => {{
+        if $condition {
+            let $iter_name = $e.$parallel();
+            $body
```

Comment:
I think this macro be be out-living its usefulness as at this point it's branching into somewhat more obscure syntax. 

I think our parallel operations are basically all `map`/`collect` so perhaps a new method could be exposed on `Compiler` which does that operation "take this list and run this closure and return the results" where the one method has internal `#[cfg]` and such for rayon vs not?